romancal/skycell/match.py

```diff
@@ -28,88 +29,167 @@ class ImageFootprint:
 
     _radec_corners: NDArray[float]
 
-    def __init__(self, radec_corners: list[tuple[float, float]]):
+    def __init__(self, radec_vertices: list[tuple[float, float]]):
         """
         Parameters
         ----------
-        radec_corners: list[tuple[float, float]]
-            four corners of the image in right ascension and declination
+        radec_vertices: list[tuple[float, float]]
+            vertices (usually the corners) of the image in right ascension and declination
         """
-        radec_corners = np.array(radec_corners)
-        if radec_corners.shape[0] != 4:
-            raise ValueError(f"need 4 corners, not {radec_corners.shape[0]}")
-        self._radec_corners = radec_corners
+        self._radec_vertices = np.array(radec_vertices)
 
     @classmethod
-    def from_gwcs(
-        cls, iwcs: WCS, image_shape: tuple[int, int] | None = None
+    def from_wcs(
+        cls,
+        wcs: WCS,
+        extra_vertices_per_edge: int = 0,
     ) -> "ImageFootprint":
-        """create an image footprint from a GWCS object (and image shape, if no bounding box is present)"""
-
-        if not hasattr(iwcs, "bounding_box") or iwcs.bounding_box is None:
-            if image_shape is None:
-                # pixel_shape is in x, y order contrary to numpy convention.
-                if hasattr(iwcs, "pixel_shape") and iwcs.pixel_shape is not None:
-                    image_shape = (iwcs.pixel_shape[1], iwcs.pixel_shape[0])
-                else:
-                    raise ValueError(
-                        "Cannot infer image footprint from GWCS object because "
-                        "`image_shape` was not specified and "
-                        "the GWCS object does not have `.bounding_box` nor `.pixel_shape`."
-                    )
+        """create an image footprint from a GWCS object (and image shape, if no bounding box is present)
+
+        Parameters
+        ----------
+        wcs: WCS :
+            WCS object
+        extra_vertices_per_edge: int :
+            extra vertices to create on each edge to capture distortion (Default value = 0)
+
+        Returns
+        -------
+        image footprint object
+        """
+
+        # if `.pixel_shape` is not defined, try deriving it from the `.bounding_box`
+        if not hasattr(wcs, "pixel_shape") or wcs.pixel_shape is None:
+            if hasattr(wcs, "bounding_box") and wcs.bounding_box is not None:
+                wcs.pixel_shape = wcs_pixel_shape_from_bbox(wcs.bounding_box)
+            else:
+                raise ValueError(
+                    "Cannot infer image footprint from WCS without `.pixel_shape` nor `.bounding_box`."
+                )
```

Comment:
I think this code is too convoluted. Is `ImageFootprint.from_wcs` run on L3 images?
 I would assume if `bounding_box` exists, then `pixel_shape` should exist as well. If it's not set, set it when the WCS is generated.

Reply:
done in d82370341bec1c0d59fdd929eac66e463dbe454c